nixos/modules/services/networking/postgrest.nix

```diff
@@ -0,0 +1,114 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}: let
+  cfg = config.services.postgrest;
+  inherit (lib) mkOption types literalExample mdDoc;
+in {
+  options.services.postgrest = {
+    enable = lib.mkEnableOption (mdDoc "postgrest");
+
+    dbURI = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = "postgres://username:password@localhost:5432/books";
+      description = mdDoc ''
+        Full Database URI containing username, password, host, port, and database name.
+      '';
+    };
+
+    postgresUser = mkOption {
+      type = types.str;
+      example = "root";
+      description = mdDoc ''
+        Username for postgres admin.
+      '';
+    };
+    postgresPassword = mkOption {
```

Comment:
Setting a password to a plain option is insecure, we should only support a file.

Reply:
@SuperSandro2000 do you know of any examples of doing this in `nixpkgs`? I'd like to have the password option be  consistent with other modules but am having trouble finding examples.

Reply:
for example ddclient does it or gitea